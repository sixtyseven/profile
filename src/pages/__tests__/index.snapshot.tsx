import React from "react"
import renderer from "react-test-renderer"
import Index from "src/pages/index"

// @ts-ignore
import MockedImage from "src/components/home/image"
// @ts-ignore
import MockedLayout from "src/components/layout/Layout"
// @ts-ignore
import MockedSEO from "src/components/common/seo"

jest.mock("src/components/home/image", () => {
  return function DummyImage(props: any) {
    return <div data-testid="image"></div>
  }
})

jest.mock("src/components/layout/Layout", () => {
  return function DummyLayout(props: any) {
    return <div data-testid="layout">{props.children}</div>
  }
})

jest.mock("src/components/common/seo", () => {
  return function DummySEO(props: any) {
    return <div data-testid="seo"></div>
  }
})

describe("Index", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Index />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
