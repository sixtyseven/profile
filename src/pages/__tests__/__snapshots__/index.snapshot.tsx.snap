// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders correctly 1`] = `
<div
  data-testid="layout"
>
  <div
    data-testid="seo"
  />
  <div
    className="row home-page"
  >
    <div
      className="col-md-4"
    >
      <div
        style={
          Object {
            "marginBottom": "1.45rem",
            "maxWidth": "300px",
          }
        }
      >
        <div
          data-testid="image"
        />
      </div>
    </div>
    <div
      className="col-md-8"
    >
      <h2>
        <Styled.code>
          Hi people
        </Styled.code>
      </h2>
      <p>
        <Styled.code>
          Welcome to my profile site.
        </Styled.code>
      </p>
      <p>
        <Styled.code>
          I am a full stack developer.
        </Styled.code>
      </p>
      <p>
        <Styled.code>
          With many years of working experience with both frontend and backend web development, I am particularly interested Modular programming, Functional programming, lean programming, scalability, OOP, SQL, NOSQL, and Unit testing etc. I have a good eye for detail. I am also a a problem solver, a good team player and enjoying collaborating with others to produce good quality work.
        </Styled.code>
      </p>
    </div>
  </div>
</div>
`;
